pkg/controllers/user/logging/utils/embedded.go

```diff
@@ -13,27 +17,32 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
 	"k8s.io/apimachinery/pkg/util/intstr"
 
 	loggingconfig "github.com/rancher/rancher/pkg/controllers/user/logging/config"
 )
 
+const (
+	failed  = "Failed"
+	running = "Running"
+)
+
 func CreateEmbeddedTarget(dep rv1beta2.DeploymentInterface, sa rv1.ServiceAccountInterface, se rv1.ServiceInterface, ro rrbacv1.RoleInterface, rb rrbacv1.RoleBindingInterface, namespace string) error {
 	// create es deployment
 	_, err := dep.Controller().Lister().Get(loggingconfig.EmbeddedESName, loggingconfig.EmbeddedESName)
 	if err != nil {
-		logrus.Infof("get embedded es error: %v", err)
 		if !apierrors.IsNotFound(err) {
 			return errors.Wrapf(err, "get deployment %s fail", loggingconfig.EmbeddedESName)
 		}
-		logrus.Infof("get embedded es error is not found")
 		// create service account, role and rolebinding
 		sc := newESServiceAccount(namespace)
 		role := newESRole(namespace)
 		roleBind := newESRoleBinding(namespace)
 
 		defer func() {
-			if err != nil {
+			if err != nil && !apierrors.IsAlreadyExists(err) {
 				sa.Delete(loggingconfig.EmbeddedESName, &metav1.DeleteOptions{})
```

Comment:
should we handle the case when remove fails? or will it always be retried anyway

Reply:
Can we return error and let the framework retry it.